HPXMLtoOpenStudio/resources/waterheater.rb

```diff
@@ -1049,61 +1113,66 @@ def self.get_default_tank_volume(fuel, num_beds, num_baths)
     end
   end
 
-  def self.get_default_recovery_efficiency(fuel, ef)
+  def self.get_default_recovery_efficiency(water_heating_system)
     # Water Heater Recovery Efficiency by fuel and energy factor
-    if fuel == HPXML::FuelTypeElectricity
+    if water_heating_system.fuel_type == HPXML::FuelTypeElectricity
       return 0.98
-    elsif ef >= 0.75
-      return 0.778114 * ef + 0.276679
+    elsif water_heating_system.energy_factor >= 0.75
+      return 0.778114 * water_heating_system.energy_factor + 0.276679
     else
-      return 0.252117 * ef + 0.607997
+      return 0.252117 * water_heating_system.energy_factor + 0.607997
     end
   end
 
-  def self.calc_ef_from_uef(uef, type, fuel_type)
+  def self.calc_ef_from_uef(water_heating_system)
     # Interpretation on Water Heater UEF
-    if fuel_type == HPXML::FuelTypeElectricity
-      if type == HPXML::WaterHeaterTypeStorage
-        return [2.4029 * uef - 1.2844, 0.96].min
-      elsif type == HPXML::WaterHeaterTypeTankless
-        return uef
-      elsif type == HPXML::WaterHeaterTypeHeatPump
-        return 1.2101 * uef - 0.6052
+    if water_heating_system.fuel_type == HPXML::FuelTypeElectricity
+      if water_heating_system.water_heater_type == HPXML::WaterHeaterTypeStorage
+        return [2.4029 * water_heating_system.uniform_energy_factor - 1.2844, 0.96].min
+      elsif water_heating_system.water_heater_type == HPXML::WaterHeaterTypeTankless
+        return water_heating_system.uniform_energy_factor
+      elsif water_heating_system.water_heater_type == HPXML::WaterHeaterTypeHeatPump
+        return 1.2101 * water_heating_system.uniform_energy_factor - 0.6052
       end
     else # Fuel
-      if type == HPXML::WaterHeaterTypeStorage
-        return 0.9066 * uef + 0.0711
-      elsif type == HPXML::WaterHeaterTypeTankless
-        return uef
+      if water_heating_system.water_heater_type == HPXML::WaterHeaterTypeStorage
+        return 0.9066 * water_heating_system.uniform_energy_factor + 0.0711
+      elsif water_heating_system.water_heater_type == HPXML::WaterHeaterTypeTankless
+        return water_heating_system.uniform_energy_factor
       end
     end
     fail 'Unexpected water heater.'
   end
 
-  def self.calc_tank_areas(act_vol)
-    pi = Math::PI
-    height = 48.0 # inches
-    diameter = 24.0 * ((act_vol * 0.1337) / (height / 12.0 * pi))**0.5 # inches
-    a_top = pi * (diameter / 12.0)**2.0 / 4.0 # sqft
-    a_side = pi * (diameter / 12.0) * (height / 12.0) # sqft
+  def self.calc_tank_areas(act_vol, height = nil)
+    if height.nil?
+      height = get_tank_height()
+    end
+    diameter = 2.0 * (UnitConversions.convert(act_vol, 'gal', 'ft^3') / (height * Math::PI))**0.5 # feet
```

Comment:
It has been verified that the only diff on CI is caused by replacing the constant number 0.1337 with calling UnitConversions method.